includes/Core/Dismissals/REST_Dismissals_Controller.php

```diff
@@ -101,9 +102,13 @@ protected function get_rest_routes() {
 				array(
 					'methods'             => WP_REST_Server::DELETABLE,
 					'callback'            => function ( WP_REST_Request $request ) {
-						$slugs = $request['data']['slugs'];
+						if ( empty( $request['data']['slugs'] ) ) {
+							// Schema validation does not catch empty object params
+							// in older versions of WP.
+							return ( new Invalid_Param_Exception( 'data' ) )->to_wp_error();
+						}
 
-						foreach ( $slugs as $slug ) {
+						foreach ( $request['data']['slugs'] as $slug ) {
```

Comment:
It seems newer versions of WP populate the properties defined in the schema but this wasn't the case for 5.3 so we had to either remove the case for testing an empty set or polyfill the validation; I chose to do the latter here.